### DISCUSSION TITLE
Pre-compiled builds ESP-IDF version

### DISCUSSION THREAD

**Discussion by ma261065** [micropython/micropython#15716] [ESP32] [answered]
Currently MicroPython on ESP32 supports ESP-IDF versions v5.0.4, v5.0.5, v5.1.2, v5.2.0, v5.2.2

I note that the pre-compiled daily builds are still compiled with the oldest version of those - v5.0.4. 

As an example https://www.micropython.org/resources/firmware/ESP32_GENERIC-20240822-v1.24.0-preview.224.g6c3dc0c0b.bin

What are the criteria for moving to later versions on the precompiled builds?

(I know that I can build my own with whatever version I want, but I'm curious why all of the fixes in later IDF versions are being left behind for the downloadable builds)

**Comment by projectgus** [comment, marked as answer]
Excellent timing to ask this question, @ma261065! Damien has just switched over to building nightlies with ESP-IDF v5.2.2 after support was added a couple of weeks ago in https://github.com/micropython/micropython/pull/15523

So the upcoming next builds *after* 20240826-v1.24.0-preview.230.ga8d1c25a1 will be IDF v5.2.2 based.

If things look good then the plan is to drop support for older ESP-IDF versions before the v1.24 release, which also means we can remove a bunch of ifdefs and conditional cleanup/workaround code, plus shrink the overall surface for possible bugs and incompatibilities. However this will depend a bit on how things progress after the nightly builds swap over (there's already https://github.com/micropython/micropython/issues/15667 but not a lot else reported yet which is encouraging).
